package/gluon-web-admin/luasrc/lib/gluon/config-mode/model/admin/remote.lua

```diff
@@ -76,29 +76,12 @@ function pw2.cfgvalue()
 end
 
 local function set_password(password)
-	local inr, inw = unistd.pipe()
-	local pid = unistd.fork()
-
-	if pid < 0 then
-		return false
-	elseif pid == 0 then
-		unistd.close(inw)
-
-		local null = fcntl.open('/dev/null', fcntl.O_WRONLY)
-		unistd.dup2(null, unistd.STDOUT_FILENO)
-		unistd.dup2(null, unistd.STDERR_FILENO)
-		if null > 2 then
-			unistd.close(null)
-		end
-
-		unistd.dup2(inr, unistd.STDIN_FILENO)
-		unistd.close(inr)
-
-		unistd.execp('passwd', {[0] = 'passwd'})
-		os.exit(127)
-	end
+	local options = {stdin = sp.PIPE,
+			 stdout = sp.DEVNULL,
+			 stderr = sp.DEVNULL}
 
-	unistd.close(inr)
+	local pid, pipe = sp.popen('passwd', {}, options)
+	local inw = pipe["stdin"]
```

Comment:
```suggestion
	local inw = pipe.stdin
```

Reply:
resolved in 6aa7d85c12b5452cd995407f20f1e984fd6c4b46